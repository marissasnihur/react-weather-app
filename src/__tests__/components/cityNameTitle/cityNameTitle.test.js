import React from 'react';
import ReactDOM from 'react-dom';

import CityNameTitle from '../../../components/cityNameTitle/cityNameTitle';

describe('cityNameTitle component...', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CityNameTitle />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should create a <h2> tag', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CityNameTitle>Edmonton</CityNameTitle>, div);
    expect(div.firstChild.nodeName).toBe('H2');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render the city name correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CityNameTitle>Edmonton</CityNameTitle>, div);
    expect(div.textContent).toBe('Edmonton');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CityNameTitle>Edmonton</CityNameTitle>, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});