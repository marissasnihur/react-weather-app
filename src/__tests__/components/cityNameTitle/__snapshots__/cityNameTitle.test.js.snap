// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cityNameTitle component... should match snapshot 1`] = `
<div>
  <h2
    class="cityNameTitle"
  >
    Edmonton
  </h2>
</div>
`;
